internal/controlplane/handlers_profile_test.go

```diff
@@ -0,0 +1,305 @@
+				{RuleType: "Type1", RuleName: "Name1"}: {Entity: minderv1.Entity_ENTITY_REPOSITORIES, RuleID: generateConsistentUUID(t, "Type1", "Name1")},
+				{RuleType: "Type2", RuleName: "Name2"}: {Entity: minderv1.Entity_ENTITY_BUILD_ENVIRONMENTS, RuleID: generateConsistentUUID(t, "Type2", "Name2")},
+			},
+			oldRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name1"}: {Entity: minderv1.Entity_ENTITY_REPOSITORIES, RuleID: generateConsistentUUID(t, "Type1", "Name1")},
+				{RuleType: "Type3", RuleName: "Name3"}: {Entity: minderv1.Entity_ENTITY_ARTIFACTS, RuleID: generateConsistentUUID(t, "Type3", "Name3")},
+			},
+			wantUnusedRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type3", RuleName: "Name3"}: {Entity: minderv1.Entity_ENTITY_ARTIFACTS, RuleID: generateConsistentUUID(t, "Type3", "Name3")},
+			},
+		},
+		{
+			name: "No unused rules in oldRules",
+			newRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name1"}: {Entity: minderv1.Entity_ENTITY_REPOSITORIES, RuleID: generateConsistentUUID(t, "Type1", "Name1")},
+				{RuleType: "Type2", RuleName: "Name2"}: {Entity: minderv1.Entity_ENTITY_BUILD_ENVIRONMENTS, RuleID: generateConsistentUUID(t, "Type2", "Name2")},
+			},
+			oldRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name1"}: {Entity: minderv1.Entity_ENTITY_REPOSITORIES, RuleID: generateConsistentUUID(t, "Type1", "Name1")},
+				{RuleType: "Type2", RuleName: "Name2"}: {Entity: minderv1.Entity_ENTITY_BUILD_ENVIRONMENTS, RuleID: generateConsistentUUID(t, "Type2", "Name2")},
+			},
+			wantUnusedRules: map[ruleTypeAndNamePair]entityAndRuleTuple{},
+		},
+		{
+			name: "Unused rules with same rule type",
+			newRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name1"}: {Entity: minderv1.Entity_ENTITY_REPOSITORIES, RuleID: generateConsistentUUID(t, "Type1", "Name1")},
+				{RuleType: "Type1", RuleName: "Name2"}: {Entity: minderv1.Entity_ENTITY_BUILD_ENVIRONMENTS, RuleID: generateConsistentUUID(t, "Type1", "Name2")},
+			},
+			oldRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name3"}: {Entity: minderv1.Entity_ENTITY_ARTIFACTS, RuleID: generateConsistentUUID(t, "Type1", "Name3")},
+			},
+			wantUnusedRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name3"}: {Entity: minderv1.Entity_ENTITY_ARTIFACTS, RuleID: generateConsistentUUID(t, "Type1", "Name3")},
+			},
+		},
+		{
+			name: "Unused old rules statuses with empty name",
+			newRules: map[ruleTypeAndNamePair]entityAndRuleTuple{
+				{RuleType: "Type1", RuleName: "Name1"}: {Entity: minderv1.Entity_ENTITY_REPOSITORIES, RuleID: generateConsistentUUID(t, "Type1", "Name1")},
+				{RuleType: "Type1", RuleName: "Name2"}: {Entity: minderv1.Entity_ENTITY_BUILD_ENVIRONMENTS, RuleID: generateConsistentUUID(t, "Type1", "Name2")},
```

Comment:
There is actually a bug in the profile handler that we can't handle rules with the _same type_ in _different entities_.  Me and @JAORMX  discussed this in discord https://discord.com/channels/1184987096302239844/1185287949240242258/1199645373791219723 (Issue to be created)

Reply:
I'm not sure "rules of the same type in different entities" makes sense (because a rule_type has an `in_entity` and rule type names are unique, IIRC).  On the other hand, I'd _also_ believe that we don't validate that correctly at the moment if you told me that.

Reply:
I'll create an issue for this. We'll look at it separately

Reply:
I'd rather defer this until we have a good enough testing framework that we can write a test to show what our behavior is, and then we can feel how we feel about what the test says.